src/action_state.rs

```diff
@@ -537,20 +539,140 @@ impl<A: Actionlike> Default for ActionState<A> {
 ///     // This component links the button to the entity with the `ActionState` component
 ///     .insert(ActionStateDriver {
 ///         action: DanceDance::Left,
-///         entity: dance_tracker,
+///         targets: dance_tracker.into(),
 ///     });
 ///```
 ///
 /// Writing your own systems that use the [`ActionStateDriver`] component is easy,
 /// although this should be reserved for cases where the entity whose value you want to check
 /// is distinct from the entity whose [`ActionState`] you want to set.
 /// Check the source code of [`update_action_state_from_interaction`](crate::systems::update_action_state_from_interaction) for an example of how this is done.
-#[derive(Component, Clone, Copy, PartialEq, Eq, Hash)]
+#[derive(Component, Clone, PartialEq, Eq)]
 pub struct ActionStateDriver<A: Actionlike> {
     /// The action triggered by this entity
     pub action: A,
     /// The entity whose action state should be updated
-    pub entity: Entity,
+    pub targets: ActionStateDriverTarget,
+}
+
+/// Represents the entities that an ``ActionStateDriver`` targets.
+#[derive(Component, Clone, PartialEq, Eq)]
+pub enum ActionStateDriverTarget {
+    /// No targets
+    None,
+    /// Single target
+    Single(Entity),
+    /// Multiple targets
+    Multi(HashSet<Entity>),
+}
+
+impl ActionStateDriverTarget {
+    /// Get an iterator for the entities targeted.
+    #[inline(always)]
+    pub fn iter(&self) -> impl Iterator<Item = &Entity> {
+        match self {
+            Self::None => ActionStateDriverTargetIterator::None,
+            Self::Single(entity) => {
+                ActionStateDriverTargetIterator::Single(std::iter::once(entity))
+            }
+            Self::Multi(entities) => ActionStateDriverTargetIterator::Multi(entities.iter()),
+        }
+    }
+
+    /// Insert an entity as a target.
+    #[inline(always)]
+    pub fn insert(&mut self, entity: Entity) {
+        *self = std::mem::replace(self, Self::None).with(entity);
```

Comment:
The mem::replace is tricky to understand (and verify) here: why use it?

Reply:
I don't want to copy the with code and make minor changes for working with a reference, but replacing a mutable ref in place is a bit of a pain

Reply:
Ah, fair enough. Just drop a comment in the source then please.